### ISSUE TITLE
Incorrect API link in twisted.web.proxy.Proxy docstring

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5316]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)| @jesstess reported|
|-|-|
|Trac ID|trac#5316|
|Type|enhancement|
|Created|2011-10-16 00:32:46Z|
|Branch|https://github.com/twisted/twisted/tree/proxy-api-link-5316|

It references ```twisted.protocols.http.HTTPChannel```, which doesn't exist. There is a ```twisted.web.http.HTTPChannel```.

<details><summary>Searchable metadata</summary>

```
trac-id__5316 5316
type__enhancement enhancement
reporter__jesstess jesstess
priority__normal normal
milestone__ 
branch__branches_proxy_api_link_5316 branches/proxy-api-link-5316
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy_documentation easy documentation
time__1318725166000000 1318725166000000
changetime__1318797343000000 1318797343000000
version__None None
owner__exarkun exarkun
cc__jesstess cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [32878]) Branching to 'proxy-api-link-5316'

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_4"></a>|@jesstess set owner to @exarkun|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [32888]) Merge proxy-api-link-5316

Author: exarkun
Reviewer: jesstess
Fixes: [#5316](https://github.com/twisted/twisted/issues/5316)

Fix a broken API reference in `twisted/web/proxy.py`.
